### ISSUE TITLE
Expand operator caching spec

### ISSUE THREAD

**Issue by shawkins** [keycloak/keycloak#30217]
### Description

The Keycloak cr has added a caching spec, but it currently only has 1 option. We can consider expanding that to additional first-class options.

### Motivation

Currently we are documenting external caching via spec additional options, these could instead by first-class options on the caching spec.

There could even be a case for expanding the caching spec sufficiently such that it could manage a dependent infinispan cr as well.

cc @ryanemerson @ahus1 @vmuzikar

**Comment by ahus1** [comment]
Thank you for bringing this up. I'd say we're ok with the current setup of `cache-remote-*` entries for now. I'd hope the `cache-embedded-mtls-*` settings would eventually vanish if we manage to store those information in Keycloak's database. 

Having the option to configure external caches in a more opinionated way than in https://www.keycloak.org/high-availability/deploy-infinispan-kubernetes-crossdc would be a benefit, as it would remove the need for "Creating the caches for Keycloak". The only variable that needs to be set there would be the `clusterName` (which might even be derived?) and the other site's site name. 

When configuring the external Infinispan, at some point we would want to know if persistent sessions are enabled, as we would then configure a limit for the number of entries, and a different number of owners. WDYT?

If we would configure the caches that way, we wouldn't allow users to configure for example a persistence for those caches, unless we add that to the Keycloak CR. 

Waiting for some other people the add there comments.

cc: @pruivo, @mhajas

**Comment by KyriosGN0** [comment]
It would be nice to have the option to specify an external cache.xml file from the CR, currently i have to use the podTemplate Spec to mount it to the container.

**Comment by ahus1** [comment]
The Keycloak CR has a configMapFile option where you can point directly to a config map.

In recent versions we added several config options so you do not need to specify an XML file in the first place. That would be the preferred way. 

If you still have the need to specify an XML file, please describe the scenario here

**Comment by KyriosGN0** [comment]
> The Keycloak CR has a configMapFile option where you can point directly to a config map.
> 
> In recent versions we added several config options so you do not need to specify an XML file in the first place. That would be the preferred way.
> 
> If you still have the need to specify an XML file, please describe the scenario here


thanks, i didn't know this works for the cache file. 
this fits my case.

**Comment by ahus1** [comment]
@KyriosGN0 - don't forget to tell my why you still need the XML.

**Comment by KyriosGN0** [comment]
> [@KyriosGN0](https://github.com/KyriosGN0) - don't forget to tell my why you still need the XML.

we want to also set the expiration max-idle node of the xml, there is not cli or env var option for that from what i see in the docs

**Comment by ahus1** [comment]
@KyriosGN0 - for what caches, and to what benefit/so satisfy what business need?

**Comment by KyriosGN0** [comment]
> [@KyriosGN0](https://github.com/KyriosGN0) - for what caches, and to what benefit/so satisfy what business need?

mainly for the local authz cache, since we control config via TF, we don't see a reason to invalidate this cache hourly 
we are trying to solve the same issue as in https://github.com/keycloak/keycloak/issues/38780 (we are running 26.2.5 and have a timeout of 3 seconds)

**Comment by ahus1** [comment]
@KyriosGN0 - the `authorization` cache is not invalidated hourly. It would only be invalidated if you change realm every hour. And there is no way a change to the cache XML can fix this for you.

**Comment by KyriosGN0** [comment]
> [@KyriosGN0](https://github.com/KyriosGN0) - the `authorization` cache is not invalidated hourly. It would only be invalidated if you change realm every hour. And there is no way a change to the cache XML can fix this for you.

when the max-idle in infispan cache is not defined is the same as not invalidating the cache ?

**Comment by ahus1** [comment]
> when the max-idle in infispan cache is not defined is the same as not invalidating the cache ?

Yes, this is how the Infinispan configuration XML works. An application can still programmatically set max idle times (and this is what we do for example on the `sessions` cache), but then the programmatically set values (usually for individual entries) override those defaults provided in the XML. 

If there would be a programmatically set limit of one hour, I would consider that a bug, and we should investigate it.
